src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2429,7 +2446,20 @@ private void DisposeIndexWriterOnError(Lazy<IndexWriteOperationBase> writeOperat
                     writeOperation.Value.Dispose();
             }
         }
+        
+        private Dictionary<string, bool> _collectionTracker;
+        
+        public IIndexedItemEnumerator EnumerateIndexedItems(IEnumerable<IndexItem> items, string collection, TransactionOperationContext indexContext,
+            IndexingStatsScope stats, IndexType type)
+        {
+            var enumerator = GetMapEnumerator(items, collection, indexContext, stats, type);
+            
+            if (IsTestRun)
+                return TestRun.CreateEnumerator(enumerator, collection, Collections.Count);
```

Comment:
For naming consistency with `CreateIndexWriteOperationWrapper` method, can we call this one `CreateEnumeratorWrapper`?